### DISCUSSION TITLE
cfclient not finding any devices when crazyradio dongle is plugged in - after using zadig to install drivers

### DISCUSSION THREAD

**Discussion by jjdike** [bitcraze/discussions#1154] [Q&A] [answered]
Hi - just starting with crazyflie.  I have the crazyradio 2.0 dongle installed on my windows 11 PC.  I used zadig to install the libusbK (v3.1.0.0) driver.  One question is that zadig shows nRF UF2 (interface 2) and nRF Serial (interface 0) for devices when I plug in the dongle.  Do both of these devices get the libusbK or libusb-win32 drivers installed?  This is what I tried so far and when I try scanning for devices with cfclient I get these messages in the cfclient command window:

WARNING:cfclient.ui.main:Cannot find a Crazyradio Dongle
Cannot find a Crazyradio Dongle
INFO:cflib.drivers.cfusb:Looking for devices....
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.

What should I do to get this correctly setup and working?  I have looked through a bunch of discussions and install instructions but am stuck.

Thanks.

**Comment by knmcguire** [comment]
Hi!

So if you just have your Crazyradio 2, you shouldn't use Zadig at all. What you are seeing is the bootloader part and you'll need to flash the crazyradio 1 emulator first according to these instructions: https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyradio-2-0/. Unforunately, this will no longer work as the windows driver for the bootloader has been overwritten with zadig.

There was a different user with this problem, so I've written instructions to recover this again. If you could give it a try and see if it works for you:

________

**How to recover Crazyradio bootloader driver on Windows.**

This is what happens when a user replaces the driver for nRF UF2 (Interface 2), with Zadig. If this happens, Windows will not be able to recognize the bootloader mode of Crazyradio as an attached hard-drive, which is necessary to flash new firmware on there.

- Press the button on the Crazyradio 2.0 while putting it in the USB port (red led flashes)
- Open up Device Manager in Windows
- Find 'nRF UF2' in any USB driver category (could be depended on which driver you selected in Zadig)
- Right-click and select 'Uninstall device'
- Select the checkbox for 'Attempt to remove the driver for this device'

In case the 'nRF Serial's driver was also replaced by serial, also uninstall this in the device manager in the same way just to be sure.

Disconnect the Crazyradio 2.0 from the USB port, and press the button while entering it back into the USB port. If everything goes well, windows should be able to open it like a drive. Also, there should be a listing of 'Crazyradio 2' under 'Portable Devices' in the device manager. Now you can follow the steps of the getting started tutorial again: https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyradio-2-0/

If not, look in the device manager if both nRF UF2 or nRF serial is listed under any of the USB driver listings and go through the steps again. If you are still experiencing issues with removing drivers, restart windows and try the steps again.

**Comment by jjdike** [comment]
Hi,

Thanks for the detailed reply and instructions.  I tried these steps
repeatedly, but wasn't getting the dongle to look like a usb drive by just
removing the driver, even after restarting.  Instead I tried updating the
driver from the device manager, selecting from drivers available on the
computer.  I picked the USB storage driver and after reinserting the
crazyradio with the button pushed in (pulsating light), the crazyradio
shows up as a portable device and shows up as a drive in the file explorer.

After downloading crazyradio2-CRPA-emulation-1.1.uf2
<https://github.com/bitcraze/crazyradio2-firmware/releases/download/1.1/crazyradio2-CRPA-emulation-1.1.uf2>
per
the crazyradio 2 instructions, I got an error message doing a cut and paste
- that the device I was trying to paste the file into didn't exist - and
the dongle light was off.  I tried reinserting the dongle with the button
pushed, getting the pulsing red light, then dragging and dropping the uf2
file, and got the same behavior.  When I check the dongle after reinserting
with the button pushed, I can see that the new uf2 file was not copied over.

I restarted the computer and retried these steps with the same result that
the dongle seems to disconnect when I do the copy.

I can see the white led light flash on when I insert the dongle without the
button pushed.  The uf2 file on the dongle is from January, while the
current uf2 file available for download is from November.  I was thinking
perhaps the cfclient would still recognize the dongle even though I hadn't
updated the uf2 file yet, but cfclient doesn't see the dongle when I scan.
I'm pretty sure I'm probably not following how this is supposed to work yet.

One question is that you mentioned that if I'm just using the crazyradio 2
I shouldn't use zadig.  But at the end of the crazyradio getting started
page,
https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyradio-2-0/
, there's a section on installing drivers.  I'm confused now as to whether
I need to install drivers or not.

Thanks again for the help.

**Reply by knmcguire** [reply]
Ah oke! I'll add updating the drivers as a possibility too

So it is normal that you don't see the file anymore in the drive, since the drag and drop is only used for flashing but not for saving the file (as it is not a drive in that sense). The white led afterwards (with not pushing the button), seems that the flashing has succeeded. 

About not using Zadig, there is a bit clarification (sorry about that). I should have said, that you shouldn't use Zadig for the bootloader (so nothing that says NRF_UF2 or NRF_serial), as that breaks that flashing procedure. But, since you are flashing a Crazyradio 1 'emulator', that does require Zadig still _after_ you have flashed it. You just have to make sure it says 'Crazyradio 2.0' on zadig and not NRF_UF2 or NRF_serial (so no pressing buttons while putting it in).

**Comment by jjdike** [comment]
Thanks - I was able to get connected to the crazyflie with the dongle.  I
have some new questions I will do in a new post.

**Reply by knmcguire** [reply]
Great! Don't forget to select an answer or write the solution and select that yourself.

**Comment by jjdike** [comment, marked as answer]
Just to close out this issue - and this still may not be totally correct - but what I was initially seeing when I plugged in the crazyflie radio 2.0 dongle were the two nRF devices when I ran Zadig, thinking I needed to run Zadig and assign the libusbK or libusb-win32 drivers in order to have the dongle show up as a usb drive for the uf2 update by drag and drop.  It seems like I couldn't get the dongle to show up as a USB drive until I assigned the winUSB driver through the windows device manager.

Also I didn't realize for a while that the dongle wouldn't be found by a scan from cfclient unless the crazyflie was also turned on.

**Reply by knmcguire** [reply]
hmm that is odd. We will keep our eye out for anybody that is experiencing the same problems then. It's not great if each windows device requires a different solution to fix this. 

Thanks for indicating what worked for you!
